nise/__main__.py

```diff
@@ -693,18 +693,19 @@ def _load_static_report_data(options):
     end_dates = []
     static_report_data = load_yaml(static_file)
     for generator_dict in static_report_data.get("generators"):
-        for _, attributes in generator_dict.items():
+        for attributes in generator_dict.values():
             start_date = get_start_date(attributes, options)
             generated_start_date = calculate_start_date(start_date)
             start_dates.append(generated_start_date)
-            if attributes.get("end_date"):
-                generated_end_date = calculate_end_date(generated_start_date, attributes.get("end_date"))
-            elif options.get("end_date") and options.get("end_date").date() != today().date():
-                generated_end_date = calculate_end_date(generated_start_date, options.get("end_date"))
-            else:
-                generated_end_date = today()
+
+            end_date = attributes.get("end_date", options.get("end_date"))
+            generated_end_date = today()
+            if end_date and end_date != today().date():
```

Comment:
pretty sure we need here what I had in my commit to enable generation of few hours of the first day of the month 
` (isinstance(options.get("end_date"), datetime.datetime) and options.get("end_date").hour != 0)`

Reply:
Investigated this with AWS and `end_date `seems to be datetime object at this point (even if specified as `--end-date 2024-03-01`) -> thus it can't equal to today().date(), meaning that this will be always applied:  `generated_end_date = calculate_end_date(generated_start_date, end_date) `
Ont the first of the month, this would mean we would get no data for current month (we are expected to use generated_end_date = today() for the first of the month)

The same holds also for dates specified in yaml using start_date: last_month, which is what our test automation uses, maning this change would break test automation on the first of the month

Reply:
will push a fix for this (the one suggested in the first comment)